lib/segment/src/index/field_index/full_text_index/inverted_index/mutable_inverted_index.rs

```diff
@@ -209,7 +209,7 @@ impl InvertedIndex for MutableInvertedIndex {
         _hw_counter: &HardwareCounterCell,
     ) -> Box<dyn Iterator<Item = PointOffsetType> + '_> {
         match query {
-            ParsedQuery::Tokens(tokens) => self.filter_has_subset(tokens),
+            ParsedQuery::Tokens(tokens) => Box::new(self.filter_has_subset(tokens)),
```

Comment:
And here?

Reply:
`filter` is a trait function which require the return type to be specified. Therefore we can't return `impl Iterator` here.

Reply:
I feel this makes the PR a bit inconsistent. Are there any perf benefits from this change?

Reply:
Trait cannot return `impl`, of course. Thanks!

@KShivendu Yes. There is a benefit, especially on chunk iterators.

Either is optimized at compile time (sort of), and box is something runtime. It dumps your data on the heap, and so it introduces some (unnecessary) indirection.